common/flatpak-utils.c

```diff
@@ -6402,12 +6402,96 @@ flatpak_progress_new (FlatpakProgressCallback progress,
                                            progress_data);
 
   g_object_set_data (G_OBJECT (ostree_progress), "callback", progress);
+  g_object_set_data (G_OBJECT (ostree_progress), "callback_data", progress_data);
   g_object_set_data (G_OBJECT (ostree_progress), "last_progress", GUINT_TO_POINTER (0));
   g_object_set_data (G_OBJECT (ostree_progress), "last_total", GUINT_TO_POINTER (0));
 
   return ostree_progress;
 }
 
+/* This should be replaced by a method in libostree; it's not possible to write
+ * it atomically using public API. This locks and unlocks many times, and is not
+ * robust to new data keys being added to OstreeAsyncProgress. Currently it only
+ * copies the keys used here by flatpak_progress_new() and
+ * FlatpakProgressCallback. */
+static void
+handle_chained_progress (OstreeAsyncProgress *chained_progress,
+                         gpointer             user_data)
+{
+  OstreeAsyncProgress *original_progress = (OstreeAsyncProgress *) user_data;
+
+  /* Sync the chained progress's state back to the original instance, to take
+   * into account any updates received while a different GMainContext was
+   * active */
+  ostree_async_progress_copy_state (chained_progress, original_progress);
```

Comment:
Note that `ostree_async_progress_copy_state()` is a new OSTree API from https://github.com/ostreedev/ostree/pull/1968, which is not yet merged. This would need a backwards compatibility fallback, or an OSTree version bump in the dependency list.

Reply:
Oops, I lost the version bump in the last force-push that I made.

Updated to include a backwards compatibility fallback in a commit which also bumps the version down again. (To be honest, I wouldn't recommend using it, we really need the libostree API, so I'd just as soon drop the last commit from this branch.)